nipoppy_cli/docs/source/user_guide/tracking.md

```diff
@@ -0,0 +1,130 @@
+```
+
+```{attention}
+Currently, only the tracker configuration with `pipeline_complete` in the `NAME` field will be used, but we are planning to extend trackers to allow multiple statuses per pipeline. Stay tuned!
+```
+
+Given a dataset with the following content in {{dpath_derivatives}}:
+```{literalinclude} ./inserts/mriqc_outputs.txt
+---
+class: no-copybutton
+---
+```
+
+Running the tracker with the above configuration will result in the imaging bagel file showing:
+```{csv-table}
+---
+file: ./inserts/mriqc_bagel.csv
+header-rows: 1
+---
+```
+
+```{note}
+If there is an existing bagel, the rows relevant to the specific pipeline, participants, and sessions will be updated. Other rows will be left as-is.
+```
+
+The `pipeline_complete` column can have the following values:
+* `SUCCESS`: all specified paths have been found
+* `FAIL`: at least one of the paths has not been found
+
+## Running a pipeline tracker
+
+### Using the command-line interface
+
+To track all available participants and sessions, run:
+```console
+$ nipoppy track \
+    --dataset-root <DATASET_ROOT> \
+    --pipeline <PIPELINE_NAME>
+```
+where `<PIPELINE_NAME>` correspond to the pipeline name as specified in the global configuration file.
+
+```{note}
+If there are multiple versions for the same pipeline in the global configuration file, use `--pipeline-version` to specify the desired version. By default, the first version listed for the pipeline will be used.
+```
+
+The tracker can also be run on a single participant and/or session at a time:
+```console
+$ nipoppy track \
+    --dataset-root <DATASET_ROOT> \
+    --pipeline <PIPELINE_NAME> \
+    --participant <PARTICIPANT_ID> \
+    --session <SESSION_ID>
+```
+
+See the [CLI reference page](<project:../cli_reference/track.md>) for more information on additional optional arguments.
+
+```{note}
+Log files for this command will be written to {{dpath_logs}}`/track`
+```
+
+### Using the Python API
+
+```python
+from nipoppy.workflows import PipelineTracker
+
+# replace by appropriate values
+dpath_root = "<DATASET_ROOT>"
+pipeline_name = "<PIPELINE_NAME>"
+
+workflow = PipelineTracker(
+    dpath_root=dpath_root,
+    pipeline_name=pipeline_name,
+)
+workflow.run()
+```
+
+See the API reference for {class}`nipoppy.workflows.PipelineTracker` for more information on optional arguments (they correspond to the ones for the [CLI](<project:../cli_reference/track.md>)).
+
+## Next steps
+
+If some participants/sessions have failed processing or have not been run yet, they should be [run again](./processing.md).
+
+Once the entire dataset has been processed with a pipeline, [Nipoppy extractors](./extraction.md) can be used to obtain analysis-ready imaging-derived phenotypes (IDPs).
```

Comment:
Not sure this is true / possible in practice. We don't have to wait for the *entire* dataset to be processed before using extractors or doing analysis. This can be iterative process.